core/server/master/src/main/java/alluxio/master/file/cmdmanager/FileSystemCommandManager.java

```diff
@@ -0,0 +1,371 @@
+    }
+
+    private static ExecutionWorkerInfo getNextAvailableWorker() throws TimeoutException {
+      // update currently available workers and get a next available worker.
+      return null;
+    }
+  }
+
+  static class CommandStatus {
+    private final long mCommandId;
+    private final String mPath;
+    private List<Task> mTasks = Lists.newArrayList();
+    private final CmdOptions mOptions;
+
+    public CommandStatus(long commandId, String path, CmdOptions options) {
+      mCommandId = commandId;
+      mPath = path;
+      mOptions = options;
+    }
+
+    public long getCommandId() {
+      return mCommandId;
+    }
+
+    public Status getCommandStatus() {
+      return null;
+    }
+
+    /*
+     * Only update bandwidth.
+     */
+    public void updateOptions(CmdOptions options) {
+      mOptions.getBandwidth().update(options.getBandwidth().getBandWidthValue());
+    }
+
+    public void addTask(Task t) {
+      mTasks.add(t);
+    }
+
+    public String getPath() {
+      return mPath;
+    }
+
+    public List<Task> getCommandTasks() {
+      return mTasks;
+    }
+
+    public String getDetailedInfo() {
+      return "";
+    }
+  }
+
+  private enum ValidationStatus {
+    New,
+    Update_Options,
+    Ignored
+  }
+
+  static class BlockIterator<T> {
+    /**
+     * Constructor to create a BlockIterator.
+     * @param filePath file path
+     */
+    public BlockIterator(String filePath) {
+    }
+
+    public List<T> getNextBatchBlocks() throws AlluxioRuntimeException {
+      return null;
+    }
+
+    /**
+     * Whether the iterator has a next complete or partial batch.
+     * @return boolean
+     */
+    public boolean hasNextBatch() {
+      return true;
+    }
+  }
+
+  private static class CmdEntry implements Journaled {
+    private final Map<String, CommandInfo>
+            mCommandPathToInfo = Maps.newHashMap();
+    private final JobIdGenerator mJobIdGenerator = new JobIdGenerator();
+
+    @Override
+    public CloseableIterator<Journal.JournalEntry> getJournalEntryIterator() {
+      return CloseableIterator.noopCloseable(mCommandPathToInfo.entrySet().stream()
+              .map(e -> Journal.JournalEntry.newBuilder().setCommandEntry(
+                      File.CommandEntry.newBuilder()
+                              .setType(e.getValue().getType().toProto())
+                              .setSrcFilePath(e.getKey()))
+                      .build())
+                      .iterator());
+    }
+
+    @Override
+    public boolean processJournalEntry(Journal.JournalEntry entry) {
+      if (entry.hasCommandEntry()) {
+        apply(entry.getCommandEntry());
+      } else {
+        return false;
+      }
+      return true;
+    }
+
+    @Override
+    public void resetState() {
+      mCommandPathToInfo.clear();
+    }
+
+    @Override
+    public CheckpointName getCheckpointName() {
+      return CheckpointName.CMD_ENTRY;
+    }
+
+    private void apply(File.CommandEntry entry) {
```

Comment:
inline this logic. Generally speaking, if a private function is only used once and is simple, inline.